p2p/host/relay/examples_test.go

```diff
@@ -0,0 +1,32 @@
+package relay_test
+
+import (
+	"context"
+
+	libp2p "github.com/libp2p/go-libp2p"
+
+	host "github.com/libp2p/go-libp2p-host"
+	routing "github.com/libp2p/go-libp2p-routing"
+)
+
+func ExampleNewRelayHost() {
+	ctx := context.Background()
+
+	libp2p.New(ctx, libp2p.EnableRelay())
```

Comment:
that's incorrect in autorelay context, you need to pass `EnableRelay(circuit.OptHop)` and the routing option.

Reply:
and `EnableAutoRelay` of course.

Reply:
I have two examples, one for relay and one for autorelay. The autorelay example includes `EnableAutoRelay` and `Routing`. I changed the examples to provide an empty list of RelayOpts, since I can't find anywhere in the code where RelayOpts are required. 

Are they required? What is the common case for relay options?

Reply:
Ok, to clarify, are there cases where libp2p consuming developers will create instances of RelayHost outside of `libp2p.New`? If so, what are they? I can add a number of examples, but my assumption was that like with AutoRelay, the supported case was only `libp2p.New`

Reply:
No, it's only through the constructor; but there are use cases where developers enable relay without autorelay.

Reply:
Yea, definitely. Check `examples_test.go` again. There are two main examples, one for Relay and one for AutoRelay, via the libp2p.New constructor. I tried to use as few optional arguments as possible, but I can add RelayOpts if necessary.

Reply:
I think we need another example that constructs the relay node with autorelay.